packages/amplify_storage_s3/lib/method_channel_storage_s3.dart

```diff
@@ -46,13 +46,13 @@ class AmplifyStorageS3MethodChannel extends AmplifyStorageS3 {
 
   @override
   Future<UploadFileResult> uploadFile(
-      {@required UploadFileRequest request}) async {
+      {required UploadFileRequest request}) async {
     try {
       final Map<String, dynamic> data =
-          await _channel.invokeMapMethod<String, dynamic>(
+      (await _channel.invokeMapMethod<String, dynamic>(
         'uploadFile',
         request.serializeAsMap(),
-      );
+      ))!;
```

Comment:
Are we certain this is non-null or should we actually be checking before the formatting step below? Ditto for the other functions updated in this file.

Reply:
Yes that's a good point.  We should do what we did for Auth.  So get a nullable map and then check if it is null: 
```

final Map<String, dynamic>? data =
          (await _channel.invokeMapMethod<String, dynamic>(
        'downloadFile',
        request.serializeAsMap(),
      ));
      if (data == null) throw nullMapFromMethodChannelException;
```